test/allow.js

```
@@ -0,0 +1,131 @@
+
+const verifiedSecretaryGraph = rdf.graph()
+verifiedSecretaryGraph.add(
+    rdf.namedNode(alice),
+    ns.acl('delegates'),
+    rdf.namedNode(agentWebId)
+  )
+const notVerifiedSecretaryGraph = rdf.graph()
+
+describe('allow() handler', () => {
+  describe('getUserId()', () => {
+    const emptyFetchDocument = () => {}
+    it('no webId present', done => {
+      let webId  // no secretary
+      let onBehalfOf = alice  // principal
+      getUserId(webId, onBehalfOf, emptyFetchDocument)
+        .then(userId => {
+          assert.equal(userId, undefined,
+            'Should return undefined when no webId is present')
+          done()
+        })
+    })
+    it('no onBehalfOf present', done => {
+      let webId = alice  // principal
+      let onBehalfOf  // no delegation header
+      getUserId(webId, onBehalfOf, emptyFetchDocument)
+        .then(userId => {
+          assert.equal(webId, userId,
+            'Should return webId when no onBehalfOf is present')
+          done()
+        })
+    })
+    it('throws 500 error if fetchDocument errors', done => {
+      let webId = agentWebId  // secretary
+      let onBehalfOf = alice  // principal
+      let fetchDocument = (url, callback) => {
+        callback(new Error('Some error while fetching'), null)
+      }
+      getUserId(webId, onBehalfOf, fetchDocument)
+        .catch(err => {
+          assert.equal(err.status, 500)
+          done()
+        })
+    })
+    it('returns principal (onBehalfOf) if secretary is verified', done => {
+      let webId = agentWebId  // secretary
+      let onBehalfOf = alice  // principal
+      let fetchDocument = (url, callback) => {
+        callback(null, verifiedSecretaryGraph)
+      }
+      getUserId(webId, onBehalfOf, fetchDocument)
+        .then(userId => {
+          assert.equal(userId, alice,
+            'Should return principal (onBehalfOf) if secretary is verified')
+          done()
+        })
+        .catch(err => {
```

You should remove this `catch` block which will suppress errors from making it back to mocha in your test.